src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpTypesThatOwnDisposableFieldsShouldBeDisposable.cs

```diff
@@ -33,21 +34,26 @@ public CSharpDisposableFieldAnalyzer(Compilation compilation)
                         model.GetSymbolInfo(assignment.Left, cancellationToken).Symbol is IFieldSymbol field &&
                         disposableFields.Contains(field))
                     {
-                        yield return field;
+                        return new[] { field };
```

Comment:
💡  Considering the number of times this returns a single element, consider copying over [`OneOrMany<T>`](https://github.com/dotnet/roslyn/blob/dc9c89a162901890c280e010f4ad11767199a6bd/src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs) from Roslyn and making that the return type.

Reply:
That's a good idea. I also considered creating an `IEnumerable<T>` implementation containing a single element and just returning that to avoid the array overhead. Which approach do you prefer?

Reply:
In the current form, it doesn't make a difference. However, I suspect a future rewrite will remove the need to use LINQ across multiple calls to `GetDisposableFieldCreations`, at which point `OneOrMany<T>` would be more efficient because it's a value type.

Reply:
Done.